### PR TITLE
mame & rom-tools 0.185

### PR DESCRIPTION
**Pull request by mbcoguno**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
* Add missing dependency.
* ~~Add Xcode 8.3 trick.~~

### CODE REVIEW

File: Formula/mame.rb [hunk 1]
```diff
@@ -17,28 +17,32 @@ class Mame < Formula
   depends_on "pkg-config" => :build
   depends_on "sphinx-doc" => :build
   depends_on "sdl2"
-  depends_on "jpeg"
+  depends_on "expat"
   depends_on "flac"
-  depends_on "portmidi"
+  depends_on "jpeg"
```
Comment (fxcoudert) [inline Formula/mame.rb]: Please do not shuffle existing dependencies

File: Formula/mame.rb [hunk 2]
```diff
@@ -17,28 +17,32 @@ class Mame < Formula
...
   depends_on "portaudio"
+  depends_on "portmidi"
+  depends_on "sqlite"
+  depends_on "utf8proc"
 
   # Needs GCC 4.9 or newer
```
Comment (fxcoudert) [inline Formula/mame.rb]: Update the comment

File: Formula/mame.rb [hunk 3]
```diff
@@ -17,28 +17,32 @@ class Mame < Formula
...
-  fails_with :gcc_4_0
-  fails_with :gcc
-  ("4.3".."4.8").each do |n|
-    fails_with :gcc => n
-  end
+  needs :cxx14
 
   def install
+    # Partial solution for Xcode 8.3 optimization issue.
+    ENV.O2 if DevelopmentTools.clang_build_version == 802
```
Comment (fxcoudert) [inline Formula/mame.rb]: Why is it a *partial* solution? And please add link to issue report

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by dunn** [comment]
Thanks so much!

**Pull request closed without merging**